man/fi_mr.3.md

```diff
@@ -810,6 +812,36 @@ or from the region.
 Providers may choose to ignore page size. This will result in a provider
 selected page size always being used.
 
+## base_mr
+
+If non-NULL, the base_mr field is used to specify the MR from which a
+sub-MR is to be created.
+
+The sub-MR inherits the memory backing the base MR and thus allows light
+weight memory registration because memory pinning can be avoided.  The
+memory range specified in mr_iov and iov_count must be fully contained in
+the memory range of the base MR.  The sub-MR can have different access
+rights from the base MR, and that means either less, same, or more access
+is allowed.  The sub-MR also has a different key from the base MR.
+
+The sub-MR should hold a reference to the base MR. When fi_close is called
+on the base MR, the call would fail if there are any outstanding sub-MRs.
+
+The base_mr field must be NULL if the FI_MR_DMABUF flag is set.
+
+## reserved_key_count
```

Comment:
I would rename.  From the viewpoint of the application, this is the number of associated memory regions.  The API also uses 'cnt' instead of 'count'.
```
sub_mr_cnt ?
assoc_mr_cnt ?
bound_mr_cnt ?
linked_mr_cnt ?
```

Reply:
Not exactly. One could reserve less keys and when used out, revert back to regular key allocation.

Reply:
The comment is about the name.  The application is creating an associated MR (not a key).  What, if anything, the provider pre-allocates isn't defined.  It could pre-allocate the fid_mr's.  I excluded using the term 'max' as part of the name because it may or may not be a max.

Reply:
Thanks, now I see your point. Since application does not need to know the pre-allocation details, it makes sense to make it a hint for intended usage instead.